tools/c7n_azure/c7n_azure/resources/app_gateway.py

```diff
@@ -33,3 +36,84 @@ class resource_type(ArmResourceManager.resource_type):
             'resourceGroup'
         )
         resource_type = 'Microsoft.Network/applicationGateways'
+
+
+@ApplicationGateway.filter_registry.register('waf')
+class ApplicationGatewayWafFilter(Filter):
+    """
+    Filter Application Gateways using WAF rule configuration.
+    State is defaulted to 'disabled'.
+
+    WAF rule group with rule ids can be found here:
+    https://learn.microsoft.com/en-us/azure/web-application-firewall/ag/application-gateway-crs-rulegroups-rules?tabs=owasp32#general-32
+
+    :example:
+
+    Return all the App Gateways which have rule '944240' disabled.
+
+    .. code-block:: yaml
+
+        policies:
+          - name: test-app-gateway
+            resource: azure.application-gateway
+            filters:
+              - type: waf
+                override_rule: 944240
+    """
+
+    schema = type_schema(
+        'waf',
+        required=[],
+        **{
+            'override_rule': {'type': 'number'},
+            'state': {'type': 'string', 'enum': ['enabled', 'disabled']}}
+    )
+
+    def process(self, resources, event=None):
+
+        filter_override_rule = self.data.get('override_rule')
+        filter_state = self.data.get('state') if self.data.get('state') else 'disabled'
```

Comment:
nit: we don't need the if/else here

```suggestion
        filter_state = self.data.get('state', 'disabled')
```